pagefaultgames/pokerogue
# [QoL] Adding challenge arrows

## What are the changes the user will see?
When going to start a challenge, left/right arrows will now show up for each challenge to show you that you can use the arrow keys to change them. When you get to the end, the appropriate arrow will disappear to let you know you've hit the end.

## Why am I making these changes?
Smewkie suggested it on discord, here we are
![image](https://github.com/user-attachments/assets/8a1f3f7c-ddee-4c39-9eba-a9d352f2d3ce)

## What are the changes from a developer perspective?
Added in some arrows based on the cursor and reverse cursor icons. They get placed onto the challenge ui handler, and the value text of the challenge (i.e. the generation number, type, on/off etc) is now centered between the two arrows. When you get to the max/min range of the challenge, it makes hides the arrows

When you go to start a challenge, once the start cursor is visible, the challenge icons will tint to show that you can't change the challenges any more. This may change to hide the cursors as per frutescens' comment, so keep an eye out for this

This also goes through the values for each challenge and bases the arrow placement on the widest text box available. This works per language, so if some languages have a longer value than others, it will still auto place the arrows and center the value text. Lastly, there's also a check for the max width of the arrow going past the edge of the challenge container. If this happens, the right arrow will be as far right as it can go, even if it means the text will overlap. 

### Screenshots/Videos
Below is a video of me going through the challenges screen and showing the arrows on display and showing it in the different languages:

https://github.com/user-attachments/assets/ada5ec37-b63e-408d-bcac-a68698364c9c

As you can see above, the position of the arrows change depending on the text in the challenges!

Below is a video of the arrows tinting when you go to hit start, and untinting when you go back to the challenge:

https://github.com/user-attachments/assets/3f1b4d07-980c-441b-b0cb-0f7e59000d09

## How to test the changes?
Download this PR and go to start a new challenge run. Try to change the values on the challenges and see how they go. This does _not_ handle other languages well yet, so that's the next thing to work on for it

## Checklist
